rawhttp-core/src/main/java/rawhttp/core/RequestLine.java

```diff
@@ -88,16 +105,38 @@ private void writeTo(OutputStream outputStream, boolean newLine) throws IOExcept
         outputStream.write(method.getBytes(StandardCharsets.US_ASCII));
         outputStream.write(' ');
 
-        String path = uri.getRawPath();
-        if (path == null || path.isEmpty()) {
-            outputStream.write('/');
+        //RFC-7230 section 5.3.3
+        if ("CONNECT".equalsIgnoreCase(method)) {
+            String host = uri.getHost();
+            int port = uri.getPort();
+
+            if (!options.allowIllegalStartLineCharacters()) {
+                if (host == null) {
+                    throw new IllegalArgumentException("URI host can not be null when CONNECT method is used");
+                } else if (port == -1) {
```

Comment:
Port `0` would probably also be illegal here because it is not a valid port, and is normally used to tell the OS that any port will do (so not appropriate for a HTTP request). So I would suggest doing `port < 1` here.